### PR TITLE
fix: rr-host booked outside of availability

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Fixes the follow issue: 

I have two rr-hosts (teampro, teamfree). Got these two personal bookings for teampro in the db (see screenshot, first two bookings). If teamfree doesn't have any availability, the 10AM slot on dez 12 (Melbourne) is not showing (that's correct). If teamfree is available the slot shows (as it should) BUT if it is teampro's round robin turn, it books teampro

Screenshot 2023-12-07 at 5 06 53 PM

The issue was the getBusyTimes didn't return the bookings because we were filtering the bookings like that: 
{
   "startTime":{"gte":"2023-12-11T23:00:00.000Z"},
   "endTime":{"lte":"2023-12-11T23:15:00.000Z"},
}

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/core/getBusyTimes.ts
Comment (alishaz-polymath) [inline packages/core/getBusyTimes.ts:261]: Do we want to keep console logs here?

**Review by alishaz-polymath — DISMISSED** [review]
LGTM, waiting on Alex to review before merging in. His context knowledge is essential here.

File: packages/core/getBusyTimes.ts
Comment (emrysal) [inline packages/core/getBusyTimes.ts:88]: Watch out, this isn't the only place this happens, https://github.com/calcom/cal.com/blob/969411041b75833ad6625d341a9d0b7ea535de43/packages/trpc/server/routers/viewer/slots/util.ts#L330

But such a good spot 💯

**Review by emrysal — CHANGES_REQUESTED** [review]

File: packages/core/getBusyTimes.ts
Comment (CarinaWolli) [reply packages/core/getBusyTimes.ts:88]: Thank you, fixed it there as well 🙏

**Review by emrysal — DISMISSED** [review]
Good to go 🔥

**Review by emrysal — APPROVED** [review]

**Pull request merged**
